runtime-components/non-repudiation-perf/src/main/scala/com/daml/nonrepudiation/perf/NonRepudiationProxyBenchmark.scala

```diff
@@ -0,0 +1,167 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.nonrepudiation.perf
+
+import java.time.Clock
+import java.util.UUID
+
+import cats.effect.{ContextShift, IO}
+import com.daml.doobie.logging.Slf4jLogHandler
+import com.daml.ledger.api.v1.command_submission_service.CommandSubmissionServiceGrpc.CommandSubmissionServiceBlockingStub
+import com.daml.ledger.api.v1.command_submission_service.{
+  CommandSubmissionServiceGrpc,
+  SubmitRequest,
+}
+import com.daml.ledger.api.v1.commands.{Command, Commands, CreateCommand}
+import com.daml.ledger.api.v1.value.{Record, RecordField, Value}
+import com.daml.nonrepudiation.client.SigningInterceptor
+import com.daml.nonrepudiation.postgresql.Tables
+import com.daml.nonrepudiation.resources.HikariTransactorResourceOwner
+import com.daml.nonrepudiation.{AlgorithmString, NonRepudiationProxy}
+import com.daml.resources.Resource
+import com.daml.resources.grpc.{GrpcResourceOwnerFactories => Resources}
+import com.daml.testing.postgresql.{PostgresAround, PostgresDatabase}
+import com.google.protobuf.duration.Duration
+import doobie.util.log.LogHandler
+import io.grpc.Server
+import io.grpc.inprocess.{InProcessChannelBuilder, InProcessServerBuilder}
+import io.grpc.protobuf.services.ProtoReflectionService
+import org.openjdk.jmh.annotations._
+import sun.security.tools.keytool.CertAndKeyGen
+import sun.security.x509.X500Name
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContext}
+
+@State(Scope.Benchmark)
+class NonRepudiationProxyBenchmark extends PostgresAround {
+
+  import NonRepudiationProxyBenchmark._
+
+  @Param(Array("100000"))
+  var commandPayloadSize: Int = _
+
+  private var stub: CommandSubmissionServiceBlockingStub = _
+  private var proxy: Resource[ExecutionContext, Server] = _
+  private var database: PostgresDatabase = _
+  private var payload: String = _
+
+  @Benchmark
+  def run(): Unit = {
+    // Generating commands adds very little noise to substantial benchmarks
```

Comment:
I checked this by first having commands ready to be shot and comparing it with this. The advantage of this approach is that we can easily create a command with a new identifier for every test, which better simulates actual production conditions.

Reply:
Because we would have to generate a lot of them in advance in order to consume them. I've tried generating 5000 commands beforehand but it consumed all of them after 3 warm-up rounds. With 100 KB payload sizes generating too many of them beforehand becomes unfeasible.

Initially I was re-using the same command over and over again, but this means that this benchmark cannot be used for longer runs to check that performance are stable with a larger number of commands being stored on the back-end. This is what I was referring to when I said that it " better simulates actual production conditions" (as in, not writing a command with the same command identifier over and over again).

Note that I was conservative in writing "very little noise", from my tests the difference is imperceptible. Consider that the cost of generating a command is negligible (especially by keeping the payload the same) compared to accessing the database.

Makes sense? Should I put if differently in the comment? I realize now this comment makes sense to me due to the fact I'm aware of the context, but is probably not very helpful for readers.